lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,227 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeLogfmt;
+
+impl Function for EncodeLogfmt {
+    fn identifier(&self) -> &'static str {
+        "encode_logfmt"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::OBJECT | kind::ARRAY,
```

Comment:
If it simplifies things, I'm happy to just include the `OBJECT` form for now and leave array of key/value pairs for later.

Reply:
I added it because I'll need it for my own use case.

For example, if I have the given fields:
`{ "ts": 2021-02-04T17:01:02, "lvl": "info", "msg": "This is a log message", "trace_id": "abcd134fg" }`

I want my final encoded string to be:

`ts=2021-02-04T17:01:02 lvl=info msg="This is a log message" trace_id=abcd134fg`

Reply:
:+1: Sorry, I mean I'd be ok with not supporting the array form: `[["level", "debug"], ["msg": "hello"], ...]`.

Reply:
Ok so I've been thinking a little bit more about this.

Like I said, I added array of key-value pairs to guarantee ordering of my fields.
An other solution would be to add an optional array parameter to enforce the ordering like so:

```
encode_logfmt(object, [ "ts", "lvl", "msg" ])
```

What do you think ?

Reply:
Ah, I see, I missed that you wanted to specify an ordering. I think your suggestion makes sense. In that case, I would expect _only_ those fields to be serialized and that they be serialized in that order.

Reply:
Ah, I was expecting the opposite :D

Basically, for my own use case, I only want to guarantee the ordering of standard fields (`ts`, `lvl`, `msg`).

So maybe do we need a third parameter (maybe a boolean ?) to let the user decide whether the fields set should be "closed" or not.

Reply:
Ah! 😄 I think that's good as long as we document the behavior given that users can achieve the other behavior via passing in just the fields they want as the object to `encode_logfmt`.